src/stores/callAPIMiddleware.js

```diff
@@ -19,7 +19,10 @@ export function callAPIMiddleware({ dispatch, getState }) {
       type: types.request,
     });
 
-    return doPerform(getState(), dispatch)
+    const request = doPerform(getState(), dispatch);
+    if (!request) return next(action);
```

Comment:
This is needed because if we use `shouldPerform()`, and it returns `false`, then the result of `doPerform()` is `undefined`, which is not a promise, so we can't use `.then()` on it.